File: android/app/src/main/java/com/cap/test/AppPlugins.java
package com.cap.test;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import android.content.Context;
import android.util.Log;

@CapacitorPlugin(name = "AppPlugins")
public class AppPlugins extends Plugin {
    @PluginMethod()
    public void test(PluginCall call){
        Context context = bridge.getContext();
        String message = call.getString("message", "");

        JSObject response = new JSObject();
        response.put("platform", "android");
        response.put("message", message);

        call.resolve(response);
    }
}

File: ios/App/App/AppPlugins.swift
import Foundation
import Capacitor

@objc(AppPlugins)
public class AppPlugins: CAPPlugin {
    @objc func test(_ call: CAPPluginCall) {
        let message: String = call.getString("message") ?? ""
        
        var response: [String: Any] = [:]
        response["platform"] = "ios"
        response["message"] = message
        
        call.resolve(response)
    }
}
